src/Sentry/Internal/RedactedHeaders.cs

```diff
@@ -0,0 +1,53 @@
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Internal;
+
+internal class RedactedHeaders : IDictionary<string, string>
+{
+    private static readonly string[] SensitiveKeys = ["Authorization", "Proxy-Authorization"];
```

Comment:
We could be a bit more aggressive there. The [Python SDK redacts a bunch of headers](https://github.com/getsentry/sentry-python/blob/b16fa5ffbad39843ebd2e9bc4ea6e91c0c9aa192/sentry_sdk/integrations/_wsgi_common.py#L39).

See [Potential suspects](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/Headers/KnownHeaders.cs).